app/products/calls/types/calls.ts

```diff
@@ -89,23 +91,31 @@ export type CallsConnection = {
 }
 
 export type ServerCallsConfig = {
-    ICEServers: string[];
+    ICEServers?: string[]; // deprecated
+    ICEServersConfigs?: ICEServersConfigs;
     AllowEnableCalls: boolean;
     DefaultEnabled: boolean;
+    NeedsTURNCredentials: boolean;
 }
 
 export type CallsConfig = {
     pluginEnabled: boolean;
-    ICEServers: string[];
+    ICEServers?: string[]; // deprecated
+    ICEServersConfigs?: ICEServersConfigs;
     AllowEnableCalls: boolean;
     DefaultEnabled: boolean;
+    NeedsTURNCredentials: boolean;
     last_retrieved_at: number;
 }
```

Comment:
@cpoile I was slightly confused by this given it's currently a superset of `ServerCallsConfig`. Is there anything that can be done to avoid the redundancy here?

Reply:
Yep, good point. We can use an intersection type to remove the redundancy.